kazu/training/train_multilabel_ner.py

```diff
@@ -0,0 +1,591 @@
+import copy
+import dataclasses
+import json
+import logging
+import pickle
+import random
+import shutil
+import tempfile
+from collections import defaultdict
+from collections.abc import Iterable
+from pathlib import Path
+from typing import Optional, Any, Union, cast
+import math
+
+import torch
+from kazu.annotation.label_studio import LabelStudioManager, LabelStudioAnnotationView
+from kazu.annotation.acceptance_test import score_sections, aggregate_ner_results
+from kazu.data import (
+    Section,
+    Document,
+    PROCESSING_EXCEPTION,
+    ENTITY_OUTSIDE_SYMBOL,
+    NumericMetric,
+    Entity,
+)
+from kazu.pipeline import Pipeline
+from kazu.steps.ner.hf_token_classification import (
+    TransformersModelForTokenClassificationNerStep,
+)
+from kazu.steps.ner.tokenized_word_processor import TokenizedWordProcessor
+from kazu.training.modelling import (
+    DistilBertForMultiLabelTokenClassification,
+    BertForMultiLabelTokenClassification,
+    DebertaForMultiLabelTokenClassification,
+)
+from torch import Tensor
+from torch.optim import AdamW
+from torch.utils.data import DataLoader
+from torch.utils.data import Dataset
+from torch.utils.tensorboard import SummaryWriter
+from tqdm import tqdm
+from transformers import (
+    PreTrainedTokenizerFast,
+    BatchEncoding,
+    PreTrainedModel,
+    get_scheduler,
+    SchedulerType,
+)
+
+from kazu.training.config import TrainingConfig
+
+logger = logging.getLogger(__name__)
+
+
+def chunks(lst: list[Any], n: int) -> Iterable[list[Any]]:
```

Comment:
There is an already an `itertools` function which does this https://docs.python.org/3/library/itertools.html#itertools.batched

Reply:
yeah you're right, maybe it was more-itertools that had it prior to that

Reply:
time to upgrade to 3.12 then 🤣